src/tests/conformance/get_data.rs

```diff
@@ -1,129 +1,237 @@
+        let blocks = vec![&SEED_BLOCKS[3], &SEED_BLOCKS[1], &SEED_BLOCKS[7]];
+        let inv_hash = blocks.iter().map(|block| block.inv_hash()).collect();
+        let query = Message::GetData(Inv::new(inv_hash));
+        let expected = blocks
+            .iter()
+            .map(|&block| Message::Block(Box::new(block.clone())))
+            .collect::<Vec<_>>();
+        let response = run_test_case(query).await.unwrap();
+        assert_eq!(response, expected);
+    }
+
+    #[tokio::test]
+    async fn blocks_1_to_8() {
+        // zcashd: pass
+        let blocks = &SEED_BLOCKS[1..=8];
+        let inv_hash = blocks.iter().map(|block| block.inv_hash()).collect();
+        let query = Message::GetData(Inv::new(inv_hash));
+        let expected = blocks
+            .iter()
+            .map(|block| Message::Block(Box::new(block.clone())))
+            .collect::<Vec<_>>();
+        let response = run_test_case(query).await.unwrap();
+        assert_eq!(response, expected);
+    }
+
+    #[tokio::test]
+    async fn non_existant_blocks() {
+        // zcashd: fails (ignores non-existant blocks).
+        let inv = Inv::new(vec![
+            InvHash::new(ObjectKind::Block, Hash::new([17; 32])),
+            InvHash::new(ObjectKind::Block, Hash::new([211; 32])),
+            InvHash::new(ObjectKind::Block, Hash::new([74; 32])),
+        ]);
+
+        let query = Message::GetData(inv.clone());
+        let expected = vec![Message::NotFound(inv)];
+        let response = run_test_case(query).await.unwrap();
+        assert_eq!(response, expected);
+    }
+
+    #[tokio::test]
+    async fn mixed_existant_and_non_existant_blocks() {
+        // Test a mixture of existant and non-existant blocks
+        // interwoven together.
+        //
+        // We expect the response to contain a Block for each
+        // existing request, and a single NotFound containing
+        // hashes for all non-existant blocks. The order of
+        // these is undefined, but since zcashd currently
+        // does not send NotFound at all, it does not matter.
+        //
+        // zcashd: fails (ignores non-existant blocks).
+
+        let non_existant_inv = vec![
+            InvHash::new(ObjectKind::Block, Hash::new([17; 32])),
+            InvHash::new(ObjectKind::Block, Hash::new([211; 32])),
+            InvHash::new(ObjectKind::Block, Hash::new([74; 32])),
+        ];
+        let blocks = SEED_BLOCKS
+            .iter()
+            .skip(3)
+            .take(non_existant_inv.len())
+            .collect::<Vec<_>>();
+
+        let expected_blocks = blocks
+            .iter()
+            .map(|&block| Message::Block(Box::new(block.clone())))
+            .collect::<Vec<_>>();
+        let expected_non_existant = Message::NotFound(Inv::new(non_existant_inv.clone()));
+
+        let mixed_inv = blocks
+            .iter()
+            .map(|block| block.inv_hash())
+            .zip(non_existant_inv)
+            .map(|(a, b)| [a, b])
```

Comment:
I feel like one should be able to `flat_map` here, but I could not please the borrow checker.

Reply:
The following should work fine?
```
        let mixed_inv = blocks
            .iter()
            .map(|block| block.inv_hash())
            .zip(non_existant_inv)
            .flat_map(|(a, b)| [a, b])
            .collect();
```

Reply:
Ah I was trying 
`.flat_map(|(a, b)| [a, b].into_iter())`
Thanks!